### DISCUSSION TITLE
Environment variables are undefined in route file

### DISCUSSION THREAD

**Discussion by rajat1saxena** [vercel/next.js#58962] [App Router] [answered]
Following is my `config/route.ts` file and when I run this via Docker, the environments variables are not found. I am injecting `SUPABASE_URL` and `SUPABASE_ANON_KEY` in the docker container via command line.

```
export async function GET(request: Request) {
    console.log(process.env.SUPABASE_URL, process.env.SUPABASE_ANON_KEY)
    return Response.json({
        url: process.env.SUPABASE_URL,
        key: process.env.SUPABASE_ANON_KEY
    })
}
```

Response
```
{}
```

What am I doing wrong?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Is the `.env` file present at build time?

This function might be getting optimized into an static JSON file.

Try adding `export const dynamic = 'force-dynamic'` to the file.

If that fixes it, you need to learn on how to opt-out from this behavior, here:

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#opting-out-of-caching

**Reply by ansh** [reply]
This worked for me! I don't understand why this is required? 

I was facing issue with Liveblocks (hopefully this helps someone in the future)
